modules/menu/menu.c

```diff
@@ -793,6 +793,17 @@ static void ua_event_handler(struct ua *ua, enum ua_event ev,
 		menu_selcall(call);
 		break;
 
+	case UA_EVENT_CALL_REDIRECT:
+		if (account_sip_autoredirect(ua_account(ua))) {
+			info("menu: redirecting call to %s\n", prm + 4);
+			ua_connect(ua, NULL, NULL, prm + 4, VIDMODE_ON);
```

Comment:
can you test if just `prm` is working ?

`prm + 4` is a bad hack and has to be fixed.

Reply:
`prm` value is of form `3xx,URI` so not likely that ua_connect would work with it.  Since we know for sure that there is three digits before , followed by the URI, I do not consider `prm + 4` a hack.  But I improved the code a bit by looking for the comma:
```
	case UA_EVENT_CALL_REDIRECT:
		uri = strchr(prm, ',') + 1;
		if (account_sip_autoredirect(ua_account(ua))) {
			info("menu: redirecting call to %s\n", uri);
			ua_connect(ua, NULL, NULL, uri, VIDMODE_ON);
		}
		else {
			info("menu: redirect call to %s\n", uri);
		}
		break;
```
Any better?